lib/child-writer/index.js

```diff
@@ -144,9 +162,13 @@ exports.write = (image, drive, options) => {
         });
       }
 
-      if (code !== EXIT_CODES.SUCCESS && code !== EXIT_CODES.VALIDATION_ERROR) {
-        return emitError(new Error(`Child process exited with error code: ${code}`));
+      // We shouldn't emit the `done` event manually here
+      // since the writer process will take care of it.
+      if (code === EXIT_CODES.SUCCESS || code === EXIT_CODES.VALIDATION_ERROR) {
```

Comment:
There are various ways the CLI process can fail, which are grouped into two categories:

- The CLI finished its task (flash & validation)
- The CLI crashed in the middle

If the CLI finished its task, then there are two possible ways out:

- The flash completed successfully (`EXIT_CODES.SUCCESS`)
- The flash completed, but the validation failed (`EXIT_CODES.VALIDATION_ERROR`)

If the CLI completes successfully, then it will emit its own `done` event, therefore we should not emit ourselves here.